app/src/main/java/com/mpieterse/stride/ui/layout/startup/viewmodels/AuthViewModel.kt

```
@@ -3,118 +3,124 @@ package com.mpieterse.stride.ui.layout.startup.viewmodels
-                        TAG, "Attempt to authenticate was a failure!"
-                    )
+                ssoClient.executeAuthenticationTransactionAsync()
+                val user = authService.getCurrentUser()
+                when (val registerState = authApi.register(user!!.email!!, user.email!!, user.uid)) {
+                    is ApiResult.Ok -> {
+                        when (val loginState = authApi.login( user.email!!, user.uid)) {
+                            is ApiResult.Ok -> {
+                                // do nothing
+                            }
+                            is ApiResult.Err -> {
+                                throw Exception()
```

detekt (1.23.8)

[warning] Exception is a too generic Exception. Prefer throwing specific exceptions that indicate a specific error case.

(detekt.exceptions.TooGenericExceptionThrown)